modules/library/main/src/main/java/org/geotools/styling/AbstractSymbolizer.java

```diff
@@ -149,9 +149,9 @@ public boolean equals(Object obj) {
         if (options == null) {
             if (other.options != null && !other.options.isEmpty())
                 return false;
-        } else if (!options.equals(other.options))
-            if (options.isEmpty() && other.options != null) 
-                return false;
+        } else if (!options.equals(other.options)) {
+            return false;
+        }
         return true;
```

Comment:
Yes, the old behavior differs, but I haven't found a good reason for it (build still works) and normal java equality does not consider null and empty as equals as equals so I just removed it.
However, someone outside of geotools/geoserver might be depending on this, it's observable behavior indeed. So I've picked your suggestion and merged it in.